### PR TITLE
Added operator-> to CircularBuffer::iterator

### PR DESCRIPTION
**Pull request by eabsi5**
Noticed these were missing. Copied from Buffer

### CODE REVIEW

**Comment by eabsi5** [comment]
Tests in a5da92605ca5d6c

File: sparta/test/CircularBuffer/CircularBuffer_test.cpp [hunk 1]
```diff
@@ -452,6 +452,44 @@ void testStatsOutput()
...
+    cm.normalize();
+    sparta::Report r1("report 1", &rtn);
+
+    sparta::StatisticSet stats(&rtn);
+    struct Entry
+    {
+        uint64_t aval;
+        bool bval;
+
+        explicit Entry(uint64_t v, bool b)
```
Comment (eabsi5) [inline sparta/test/CircularBuffer/CircularBuffer_test.cpp]: I think I was originally going to have default values in there :)

File: sparta/sparta/resources/CircularBuffer.hpp [hunk 2]
```diff
@@ -265,6 +265,18 @@ namespace sparta
                 sparta_assert(isValid(), "Iterator is not valid for dereferencing");
                 return circularbuffer_entry_->data;
             }
+            value_type* operator->() {
```
Comment (criusx) [inline sparta/sparta/resources/CircularBuffer.hpp]: opening brace on the next line to match with existing style

Comment (eabsi5) [reply sparta/sparta/resources/CircularBuffer.hpp]: Fixed in 2295bdccdb872

**Pull request merged**